engine/table/src/main/java/io/deephaven/engine/table/impl/SourceTable.java

```diff
@@ -363,7 +363,7 @@ protected void destroy() {
     @Override
     @NotNull
     public Stream<PerformanceEntry> sourceEntries() {
-        if (updateSourceRegistrar != null) {
+        if (updateSourceRegistrar != null && locationsInitialized) {
```

Comment:
My only comment is that other bits of code in this class do a null check on `locationChangePoller` instead since `locationsInitialized` might be true but the former could still be null. The javadoc would seem to impl sourceEntries returns an empty stream in static contexts.